kernel/src/main/scala/me/rexim/morganey/ast/LambdaTerm.scala

```diff
@@ -82,8 +51,24 @@ case class LambdaFunc(parameter: LambdaVar, body: LambdaTerm) extends LambdaTerm
 
   override val freeVars: Set[String] = body.freeVars - parameter.name
 
+  private[ast] def nestedFunctions(): (List[LambdaVar], LambdaTerm) = {
+    @tailrec
+    def go(func: LambdaFunc, acc: List[LambdaVar]): (List[LambdaVar], LambdaTerm) =
```

Comment:
This change will throw a `MatchError` at runtime, if this method was called from an instance with the wrong type. This method is only allowed to be called on instances of `LambdaFunc`.